src/Abstractions/NexusMods.Abstractions.Loadouts.Synchronizers/SyncActionGroupings.cs

```diff
@@ -0,0 +1,37 @@
+using System.Collections.Concurrent;
+using NexusMods.Abstractions.Loadouts.Synchronizers.Rules;
+
+namespace NexusMods.Abstractions.Loadouts.Synchronizers;
+
+/// <summary>
+/// A grouping of actions that can be performed on a file. The SyncTree has a `Actions`
+/// field and that field is used to put the items in the tree into these specific groups.
+/// </summary>
+public class SyncActionGroupings
+{
+    private ConcurrentDictionary<Actions, ConcurrentBag<SyncTreeNode>> _groupings = new();
+    
+    /// <summary>
+    /// Gets the group of nodes that have the specified action.
+    /// </summary>
+    /// <param name="action"></param>
+    public IReadOnlyCollection<SyncTreeNode> this[Actions action] => _groupings.GetOrAdd(action, static _ => []);
+    
+    /// <summary>
+    /// Adds a node to the groupings based on the actions it has.
+    /// </summary>
+    /// <param name="node"></param>
+    public void Add(SyncTreeNode node)
+    {
+        foreach (var flag in Enum.GetValues<Actions>())
+        {
+            if (node.Actions.HasFlag(flag))
+                AddOne(flag, node);
+        }
+    }
+    
+    private void AddOne(Actions flag, SyncTreeNode node)
+    {
+        _groupings.GetOrAdd(flag, static _ => []).Add(node);
+    }
+}
```

Comment:
Good point, I haven't yet added any parallel code around this but it's fairly trivial to do so. My goal was to try and get this code done so it would un-block people while I'm out on holiday. I plan on using the next few days to do performance improvements and other small feature additions separate PRs